### PR TITLE
Add Ability to Test a Downloaded Go CLI

### PR DESCRIPTION
**Pull request by dubeejw**
- Use a downloaded Go CLI with automationed test suite if property is specificed in whisk.properties
- Get downloaded Go CLI path
- Update WskCLI.java, and Wsk.scala to run automation against a downloaded Go CLI

### CODE REVIEW

File: tests/src/common/WskCli.java [hunk 1]
```diff
@@ -982,9 +989,9 @@ public RunResult cli(boolean verbose, int expectedExitCode, String... params) th
     public RunResult cli(boolean verbose, int expectedExitCode, File workingDir, String... params) throws IllegalArgumentException, IOException {
         String[] cmd;
         if (usePythonCLI) {
-            cmd = WhiskProperties.useCliDownload() ? new String[] { getDownloadedCliPath() } : new String[] { WhiskProperties.python, new File(cliPath).toString() };
+            cmd = WhiskProperties.useCLIDownload() ? new String[] { WhiskProperties.python, getDownloadedCLIPath() } : new String[] { WhiskProperties.python, new File(cliPath).toString() };
```
Comment (dubeejw) [inline tests/src/common/WskCli.java]: Not sure what the point creating a new file then printing the string value of the path is here: new File(cliPath).toString().

**Comment by dubeejw** [comment]
As a smoke tests for this I placed the CLI in my home folder ./local/bin/go-cli, set use.cli.download to true in whisk.properties, and the tests are passing.

**Comment by dubeejw** [comment]
Please review.

**Comment by dubeejw** [comment]
IPG 263.

File: tests/src/common/WskCli.java [hunk 2]
```diff
@@ -107,8 +107,8 @@ public String getSubject() {
     public boolean checkExists() {
         File dir;
 
-        if (WhiskProperties.useCliDownload()) {
-            String binary = getDownloadedCliPath();
+        if (WhiskProperties.useCLIDownload()) {
```
Comment (mdeuser) [inline tests/src/common/WskCli.java]: Could all of the wsk CLI path/dir information be settled withing the WskCli constructor?  After the constructor is called, the cliPath and cliDir would be set appropriately (python vs go, downloaded cli vs built).  That way, there would not be a need for other conditional usePython or useCLIDownload checks in other methods.

**Comment by mdeuser** [comment]
@dubeejw - review complete; one comment added

**Comment by dubeejw** [comment]
@mdeuser, ready for another review.

**Comment by mdeuser** [comment]
@dubeejw - refactor of CLI path processing looks good.

**Comment by rabbah** [comment]
Merged as https://github.com/openwhisk/openwhisk/commit/4f185e6d2c2137812b6df7ddc663e15cf12bbc16.

**Pull request closed without merging**